src/renderer/components/Input/InputDelegate.vue

```diff
@@ -0,0 +1,362 @@
+    isFocused () {
+      if (this.isFocused && this.hasSuggestions) {
+        this.openDropdown()
+      }
+    },
+
+    inputValue () {
+      this.dropdownValue = null
+      if (this.isFocused && this.hasSuggestions) {
+        this.openDropdown()
+      }
+    }
+  },
+
+  mounted () {
+    if (this.value) {
+      this.updateInputValue(this.value)
+    }
+  },
+
+  methods: {
+    getHelperText () {
+      return (!this.$refs.dropdown || !this.$refs.dropdown.isOpen) ? this.helperText : ''
+    },
+
+    blur () {
+      this.$refs.input.blur()
+    },
+
+    focus () {
+      this.$refs.input.focus()
+    },
+
+    onBlur (evt) {
+      // Verifies that the element that generated the blur was a dropdown item
+      if (evt.relatedTarget) {
+        const classList = evt.relatedTarget.classList || []
+        const isDropdownItem = _.includes(classList, 'MenuDropdownItem__button')
+
+        if (!isDropdownItem) {
+          this.closeDropdown()
+        }
+      } else if (this.$refs.dropdown.isOpen) {
+        this.closeDropdown()
+      }
+
+      this.isFocused = false
+
+      // If the user selects a suggestion and leaves the input
+      if (this.dropdownValue) {
+        this.onEnter()
+      }
+    },
+
+    onDropdownSelect (value) {
+      this.model = value
+      this.$nextTick(() => this.closeDropdown())
+    },
+
+    onFocus () {
+      this.isFocused = true
+      this.$emit('focus')
+    },
+
+    onEnter () {
+      if (!this.dropdownValue) return
+      this.model = this.dropdownValue
+
+      this.$nextTick(() => {
+        this.closeDropdown()
+        this.$refs.input.setSelectionRange(this.inputValue.length, this.inputValue.length)
+      })
+    },
+
+    onEsc () {
+      this.dropdownValue = null
+      this.closeDropdown()
+    },
+
+    onKeyUp () {
+      const next = this.dropdownValue ? this.suggestionsKeys.previous() : this.suggestionsKeys.current()
+      this.__setSuggestion(next)
+    },
+
+    onKeyDown () {
+      const next = this.dropdownValue ? this.suggestionsKeys.next() : this.suggestionsKeys.current()
+      this.__setSuggestion(next)
+    },
+
+    onDecodeQR (value, toggle) {
+      const address = this.qr_getAddress(value)
+
+      // Check if we were unable to retrieve an address from the qr
+      if ((address === '' || address === undefined) && address !== value) {
+        this.$error(this.$t('MODAL_QR_SCANNER.DECODE_FAILED', { data: value }))
+      }
+
+      this.model = this.$store.getters['delegate/byAddress'](address).username
+      this.$nextTick(() => {
+        this.closeDropdown()
+        toggle()
+      })
+    },
+
+    closeDropdown () {
+      this.$refs.dropdown.close()
+    },
+
+    openDropdown () {
+      this.$refs.dropdown.open()
+    },
+
+    async updateInputValue (value) {
+      this.inputValue = value
+      // Inform Vuelidate that the value changed
+      this.$v.model.$touch()
+    },
+
+    __setSuggestion (value) {
+      if (!this.hasSuggestions) return
```

Comment:
Try to keep if statements consistent with curly brackets as it helps with readability